### DISCUSSION TITLE
ZUI in docker requirements

### DISCUSSION THREAD

**Discussion by apella12** [zwave-js/zwave-js-ui#3322] [Q&A] [answered]
Total noob question, but I have ZUI running in Docker.  Do I need to install nodejs on the bare metal or is this part of the docker package?  I did install the nodejs repository, but just wondering.

Also the default sleep duration is 250 ms, shouldn't the unit on that parameter in ZUI be ms with 250 ms steps up to the 5 second max?

**Comment by robertsLando** [comment, marked as answer]
> Do I need to install nodejs on the bare metal or is this part of the docker package? I did install the nodejs repository, but just wondering.

Nope docker container ships everything that is needed in order to run zui so no need to install anything on your machine (except docker of course).

> Also the default sleep duration is 250 ms, shouldn't the unit on that parameter in ZUI be ms with 250 ms steps up to the 5 second max?

That setting parameter is just to change it, if you leave it empty it will use the default value (now it's 250ms but in future it could be changed to be lower or higher on driver side, this is why is kept empty)
